### PR TITLE
[stf-collect-logs] Add a role for log collection

### PR DESCRIPTION
**Pull request by elfiesmelfie**

### CODE REVIEW

**Comment by elfiesmelfie** [comment]
Jenkins says this doesn't break anything, but it needs some additional testing, I'm sure.

File: build/stf-collect-logs/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+- name: "Get builds"
+  ansible.builtin.shell:
+    cmd: |
+      echo "*** [INFO] Showing oc get builds" > {{ logfile_dir }}/post_oc_get_builds.log 2>&1
```
Comment (elfiesmelfie) [inline build/stf-collect-logs/tasks/main.yml:5]: Immediately, I see that I didn't add a logfile_dir, and that should actually have broken something.

File: build/run-ci.yaml [hunk 2]
```diff
@@ -6,4 +6,8 @@
   tasks:
   - name: Run the STF CI system
     import_role:
-        name: stf-run-ci
+      name: stf-run-ci
+          
+  - name: Collect the logs
+    import_role:
+      name: stf-collect-logs
```
Comment (elfiesmelfie) [inline build/run-ci.yaml:13]:
```suggestion
      name: stf-collect-logs
    vars:
      logfile_dir: "{{ playbook_dir }}/"
```

**Review by csibbitt — APPROVED** [review]
"that should actually have broken something."  Is this an open concern that jenkins failed to report a failure where you thought it should? Or did you figure out what happened?

**Comment by elfiesmelfie** [comment]
> "that should actually have broken something." Is this an open concern that jenkins failed to report a failure where you thought it should? Or did you figure out what happened?

I'm only getting around to looking at it today. It'll be easier to test now that we have one of the Zuul jobs also running.

**Comment by elfiesmelfie** [comment]
recheck

The Zuul errors was related to CRC not starting. I'm assuming that was an intermittent failure for now.

**Review by leifmadsen — APPROVED** [review]

**Pull request merged**